lib/facility_factory.rb

```diff
@@ -0,0 +1,46 @@
+class FacilityFactory
+attr_reader :facilities
+
+  def initialize()
+    @facilities = []
+  end
```

Comment:
I like your individual source methods. A next step might be to have a method called in initialize (`load_source`?) to determine the source and kick off the correct `add_xx_source` method so you can go right ahead and `create_facilities` when you're ready. Great work!